src/script/lua_api/l_env.cpp

```diff
@@ -880,6 +880,21 @@ int ModApiEnvMod::l_find_node_near(lua_State *L)
 	return 0;
 }
 
+static void checkArea(v3s16 &minp, v3s16 &maxp)
+{
+	auto volume = VoxelArea(minp, maxp).getVolume();
+	// Volume limit equal to 8 default mapchunks, (80 * 2) ^ 3 = 4,096,000
+	if (volume > 4096000) {
+		throw LuaError("Area volume exceeds allowed value of 4096000");
+	}
+
+	// Clamp to map range to avoid problems
+#define CLAMP(arg) core::clamp(arg, (s16)-MAX_MAP_GENERATION_LIMIT, (s16)MAX_MAP_GENERATION_LIMIT)
```

Comment:
Regardless of the boundaries you choose, I think Minetest should output a warning here if the clamping happens, to detect misbehaving mods, mapgens, and players.